modules/cloudflare/dns/variables.tf

```diff
@@ -0,0 +1,68 @@
+variable "domain" {
+  description = "The top-level domain name to hold the records."
+  type        = string
+}
+
+variable "account_id" {
+  description = "The ID of the Cloudflare account."
+  type        = string
+}
+
+variable "create_zone" {
+  description = "Whether to create the zone. Defaults to `true`."
+  type        = bool
+  default     = true
+}
+
+variable "records" {
+  description = <<EOT
+    List of DNS records for the domain.
+
+    • `name`     - (Optional) The name of the record. Defaults to "@" (i.e. an apex record).
+    • `ttl`      - (Optional) The TTL of the record. Defaults to `default_ttl`.
+    • `type`     - (Required) The record type.
+    • `content`  - (Required) The content of the record.
+    • `priority` - (Optional) The priority of the record.
+    • `proxied`  - (Optional) Whether to use Cloudflare’s origin protection. Defaults to `false`.
+  EOT
+  type = map(object({
+    name     = optional(string, "@")
+    ttl      = optional(number)
+    type     = string
+    content  = string
+    priority = optional(number)
+    proxied  = optional(bool, false)
+  }))
+}
+
+variable "default_ttl" {
+  description = "Default TTL for DNS records. Defaults to 1, which means “automatic”."
+  type        = number
+  default     = 1
```

Comment:
Should we have some validation on this number e.g. can't be less than `0`/`1`? Not sure what the validation should be though - <https://developers.cloudflare.com/dns/manage-dns-records/reference/ttl/>

Reply:
You're right, I could add some validation for this. This shouldn't be provided for proxied records so I'd need to check how that behaves, and for non-proxied records it can either be `1` (auto, 300 secs) or a number between `60` and `86400`. I will do a bit more digging on behaviour, but I've never provided a TTL for any Cloudflare records in the past, just left it on Auto